k8s/tools/deploy-pubsub-gke.sh

```diff
@@ -0,0 +1,278 @@
+fi
+
+echo "Deploying cluster to project $DEVSHELL_PROJECT_ID"
+
+# Setup appropriate directories and copy of deployment templates and Turbinia config
+echo "Copying over template deployment files to $DEPLOYMENT_FOLDER"
+mkdir -p $DEPLOYMENT_FOLDER
+cp gcp-pubsub/* $DEPLOYMENT_FOLDER
+cp ../turbinia/config/turbinia_config_tmpl.py $DEPLOYMENT_FOLDER/$TURBINIA_CONFIG
+
+# Deploy cloud functions
+if [[ "$*" != *--no-cloudfunctions* ]] ; then
+  echo "Deploying cloud functions"
+  gcloud -q services --project $DEVSHELL_PROJECT_ID enable cloudfunctions.googleapis.com
+  gcloud -q services --project $DEVSHELL_PROJECT_ID enable cloudbuild.googleapis.com
+
+  # Deploying cloud functions is flaky. Retry until success.
+  while true; do
+    num_functions="$(gcloud functions --project $DEVSHELL_PROJECT_ID list | grep task | grep $REGION | wc -l)"
+    if [[ "${num_functions}" -eq "3" ]]; then
+      echo "All Cloud Functions deployed"
+      break
+    fi
+    gcloud --project $DEVSHELL_PROJECT_ID -q functions deploy gettasks --region $REGION --source ../tools/gcf_init/ --runtime nodejs14 --trigger-http --memory 256MB --timeout 60s
+    gcloud --project $DEVSHELL_PROJECT_ID -q functions deploy closetask --region $REGION --source ../tools/gcf_init/ --runtime nodejs14 --trigger-http --memory 256MB --timeout 60s
+    gcloud --project $DEVSHELL_PROJECT_ID -q functions deploy closetasks  --region $REGION --source ../tools/gcf_init/ --runtime nodejs14 --trigger-http --memory 256MB --timeout 60s
+  done
+fi
+
+# Deploy Datastore indexes
+if [[ "$*" != *--no-datastore* ]] ; then
+  echo "Enabling Datastore API and deploying datastore index"
+  gcloud --project $DEVSHELL_PROJECT_ID -q services enable datastore.googleapis.com
+  gcloud --project $DEVSHELL_PROJECT_ID -q datastore indexes create ../tools/gcf_init/index.yaml
+fi
+
+# Create GKE cluster and authenticate to it
+if [[ "$*" != *--no-cluster* ]] ; then
+  echo "Enabling GCP Compute and Container APIs"
+  gcloud -q services --project $DEVSHELL_PROJECT_ID enable compute.googleapis.com
+  gcloud -q services --project $DEVSHELL_PROJECT_ID enable container.googleapis.com
+  echo "Creating cluser $CLUSTER_NAME with $CLUSTER_NODE_SIZE node(s) configured with machine type $CLUSTER_MACHINE_TYPE and disk size $CLUSTER_MACHINE_SIZE"
+  gcloud beta container clusters create $CLUSTER_NAME --machine-type $CLUSTER_MACHINE_TYPE --disk-size $CLUSTER_MACHINE_SIZE --num-nodes $CLUSTER_NODE_SIZE --master-ipv4-cidr $VPC_CONTROL_PANE --network $VPC_NETWORK --zone $ZONE --shielded-secure-boot --no-enable-master-authorized-networks  --enable-private-nodes --enable-ip-alias  --scopes "https://www.googleapis.com/auth/cloud-platform" --labels "turbinia-infra=true"
+else
+  echo "--no-cluster specified. Authenticating to pre-existing cluster $CLUSTER_NAME"
+fi
+
+# Authenticate to cluster
+gcloud container clusters get-credentials $CLUSTER_NAME --zone $ZONE
```

Comment:
So in general setting `gcloud config set project <PROJECT_NAME>` should be enough to run all commands against that project. the `--project` is to specify a different project but just in case user error I went ahead and added --project for every gcloud command available

Reply:
Cool, thanks for fixing that up.  I think it's good to have this set explicitly since it doesn't look like we actually check the project in the case that variable is already set, so it would be possible to run gcloud commands on different projects in that case.

Reply:
Ohh okay I see/makes sense. Cool then should be g2g in this part